lib/awesome_spawn.rb

```diff
@@ -164,7 +164,7 @@ def launch(env, command, spawn_options)
   def parse_command_options(command, options)
     options = options.dup
     params  = options.delete(:params)
-    env = options.delete(:env) || {}
+    env = (options.delete(:env) || {}).map { |n, v| [n.to_s, v.to_s] }.to_h
```

Comment:
Agree with rubocop here:

```suggestion
    env = (options.delete(:env) || {}).to_h { |n, v| [n.to_s, v.to_s] }
```

Reply:
Actually, no, `.to_h` is Ruby 2.6+ and this gem still claims support for Ruby 2.5

Reply:
`n.to_s` is this even valid?
```
$ FOO=foo env|grep FOO
FOO=foo
$ 12=foo env|grep FOO
bash: 12=foo: command not found
```

Reply:
I was thinking symbol support, but yeah, the .to_s allows us to pass all kinds of things that shouldn't be allowed.  This is part of the reason I'm on the fence about it.

Reply:
Oh you can't pass symbols to Kernel.spawn either? (checking...)

Reply:
Sure enough
```
>> Kernel.spawn({:FOO => "bar"}, "echo $FOO")
(irb):4:in `spawn': no implicit conversion of Symbol into String (TypeError)
```

Reply:
Actually, there is one use case that this breaks, now that I'm digging in more.  So, I think this is a bug

Kernel.spawn _does_ allow passing `nil` as a value to unset the ENV var in the child process.  This PR breaks that by changing it to a blank set alwas.  See the difference here:

```
> Kernel.spawn({"FOO" => "xxxx"}, "echo ===== && echo $(env | grep FOO) && echo =====")
=====
FOO=xxxx
=====
> Kernel.spawn({"FOO" => ""}, "echo ===== && echo $(env | grep FOO) && echo =====")
=====
FOO=
=====
> Kernel.spawn({"FOO" => nil}, "echo ===== && echo $(env | grep FOO) && echo =====")
=====

=====
```

Reply:
ok. have tests for `nil`, `""`,  and numbers. `ENV.include?("FOO")` bug should be fixed.
The `to_s` was concentrating on symbol keys.
Ruby 2.7 is no longer supported, going across to the new `to_h` could work.

Having `number => value` work seems minor, but holler if you want me to try and fix.